src/moonbeam/common.js

```diff
@@ -139,50 +114,34 @@ export const scheduleTask = async ({
         oakAdapter,
         destinationChainAdapter: moonbeamAdapter,
         taskPayloadExtrinsic,
-        scheduleFeeLocation: moonbeamDefaultAsset.location,
-        executionFeeLocation: moonbeamDefaultAsset.location,
-        scheduleAs: u8aToHex(keyringPair.addressRaw),
+        scheduleFeeLocation: oakAsset.location,
+        executionFeeLocation: moonbeamAsset.location,
         keyringPair: moonbeamKeyringPair,
+        caller: moonbeamAdapter,
+        callerXcmFeeLocation: oakAsset.location,
     }, schedule);
-    const listenEventsPromise = listenEvents(oakApi, 'automationTime', 'TaskScheduled', 60000);
-    const results = await waitPromises([sendExtrinsicPromise, listenEventsPromise]);
-    const { foundEvent: taskScheduledEvent } = results[1];
-    const taskId = getTaskIdInTaskScheduledEvent(taskScheduledEvent);
-    console.log(`Found the event and retrieved TaskId, ${taskId}`);
-
-    const executionTime = scheduleActionType === ScheduleActionType.executeOnTheHour
-        ? nextExecutionTime : moment().valueOf() / 1000;
-    const timeout = calculateTimeout(nextExecutionTime);
-
-    console.log(`\n4. Keep Listening events on ${parachainName} until ${moment(executionTime * 1000).format('YYYY-MM-DD HH:mm:ss')}(${executionTime}) to verify that the task(taskId: ${taskId}) will be successfully executed ...`);
 
-    const listenEventsResult = await listenEvents(oakApi, 'automationTime', 'TaskTriggered', { taskId }, timeout);
+    const executionTime = scheduleActionType === ScheduleActionType.executeImmediately ? (Math.floor(moment() / 1000) + 60) : nextExecutionTime;
+    const [, { taskId, messageHash }] = await waitPromises([sendExtrinsicPromise, listenXcmpTaskEvents(oakApi, executionTime)]);
```

Comment:
Merge the listening logic for TaskScheduled and TaskTriggered events, addressing an issue where waiting for extrinsic finalization was blocking the listening of TaskTriggered, thus causing the TaskTriggered event to be inaccessible.

Reply:
Okay, nice.